### ISSUE TITLE
According to new Ban Wave

### ISSUE THREAD

**Issue by i-Grey** [Masuzu/ZooeyBot#1120]
There're plenty of people whom got banned in today BAN WAVE.
I though most of them are VIRAMATE users but I'm not quite sure that we still can considering ZooeyBot as safe or not. Some VIRAMATE'er got banned, someone not.

Although, there's still some people whom claimed himself that he didn't use any 3rd party tools at all and still got banned.

Could you confirm about this issue @Masuzu

**Comment by giauten** [comment]
Everyone with Extensions that access to the game has been banned, warning with adblocker
Zoi is using Extensions to access game's data so
I think this will be the end

**Comment by Poko302** [comment]
I have been using Viramate until the developer announcing to stop develop Viramate then I uninstalled it.
I did not get ban but I get warning pop up about they knew that I once used 3rd party tool to change UI or other aspects of the game. 
I think we need to wait for @Masuzu to confirm about the new detection tool first.

**Comment by krypticsalt** [comment]
We have to wait on Masuzu, the warning does say 'an unauthorized tool has been detected making changes to the user interface or other aspects of the game', so I dont know if Zooey would be detected. Zooey also doesnt have a fixed chrome store ID so that wouldnt work for detection too. Maybe Masuzu knows about more and can look into it?

**Comment by i-Grey** [comment]
I didn't use Viramate and I'm still don't get any Warning Pop-up at all. What about you guys?

**Comment by Hildibrand** [comment]
**As a warning, all I'm about to say is stuff I've read on Discord and chats, so take it with a grain of salt because it may be people coping or panicking.**
As far as I've seen, even stuff like AdBlocker can trigger a false positive, since it modifies HTML code and it, supposedly, can be used to skip animations. I've been using viramate to farm Xeno, Peace of Pie and doing daily hard+omega, but not running it for more than one hour straight. I use Viramate when I'm not using ZooeyBot. I still got banned until the 22nd.

Apparently there are some japanese users that have been banned for using iOS so maybe Cygames just fucked up. I mean, if ZooeyBot was really flagged as botting, we would have to be banned forever, right? Not temporarily. I suppose it means Cygames can't really tell what's botting and what's not so they're banning any extension that messes with HTML. Just my thoughts.

RaidFinder has been lowering its functionality: at first you could click "join" and inmediately get to summon page. After that, it got you to "join raid" page with the raid code written. Right now, all it does is copying the code to your clipboard. So maybe it's the instant shortcuts that are flagged (even Mazusu said that the way ZooeyBot refills pot could be reworked). I figure instant pot refills may be a little suspicious.

But yeah, let's see what Masuzu has to say about it. Maybe they got banned as well.

**Comment by smallqp** [comment]
Many people are being Ban today, and I am one of them.
I am a user of zooey and sarasa
I have used Viramate for only a few minutes, but have not deleted it since.
I don't know where the reason is, so everyone is still careful.

Google translation

**Comment by Runaris** [comment]
the game can now detect extensions, anything it doesn't recognize or is whitelisted = banned for 7 days

adblock seems safe, users only appear to be warned.

**Comment by rockysilver** [comment]
I only got a warning, no ban thankfully.

I farmed all the items in this events over 2 days (4 hours per day); and 20 boxes the last event. 

I have viramate active, but did not use viramate with zooey. 

Anyone else used zooey a lot lately? Any bans?

**Comment by himappe** [comment]
maybe safe only using zoooeybot that  is true 
>>
Zooey does not inject any Javascript code or HTML contents tampering with the original GBF webpage.

**Comment by YumeHaru** [comment]
I can confirm some facts since I did some test:
- You get a warning with ad blocker even on a mobile phone so.
- One of my accounts got banned 7 days but it's probably due to the fact I didn't get the warning since I was lagging like hell on the login page during the ban wave, so I like an idiot I refreshed login page while I should have disabled Viramate before login in again, imo.

 I think if they suspected me botting, I would had a heavier punishment.
For info, the account who got banned did not bot since the anniversary event same for the others accounts and I was barely playing GBF since then but I kept Viramate enabled all this time till today.

**Comment by ErogeSeiyuu** [comment]
slime alt: Not banned and also no warning despite having viramate turned on while sliming (but no hotkeys used since it's just half pot slave)
 
main acc: banned. I think it's due to using hotkeys since I haven't been using zoibot since Friday, but I farmed NezhaHL and UBHN with viramate hotkeys on.

**Comment by YumeHaru** [comment]
In game announcement that you can see on the "News" page
https://cdn.discordapp.com/attachments/321510644392263680/567365383166885908/ss.JPG

**Comment by SunStrider8420** [comment]
> I only got a warning, no ban thankfully.
> 
> I farmed all the items in this events over 2 days (4 hours per day); and 20 boxes the last event.
> 
> I have viramate active, but did not use viramate with zooey.
> 
> Anyone else used zooey a lot lately? Any bans?

Used Zooeybot, Sarasabot & Viramate alot.
No ban. 

Just a warning.

**Comment by zappky** [comment]
All i do is look at my grid, go back to homepage, poof, Main account got banned without warning. I wasnt even doing anything,but only have viramate still enabled.

While my two alts acc got warning even though the browsers used for those two are clean of "bad" extension,which only have adobe and google drive extension.

**Comment by eddierayo156** [comment]
I've used zooeybot for the events since new years on two account as well as sarasa bot on magfests. I got a warning when I first opened the granblue app on my android. No warning on my alt when I opened it on google chrome on my pc.

**Comment by GBFUSER1993** [comment]
@eddierayo156 have you use viramate?

**Comment by eddierayo156** [comment]
I download it once last year but deleted it after a few minutes.

**Comment by YumeHaru** [comment]
If you guys read the link I posted above, it says that everyone got that warning with no exceptions and a long post was made on Reddit by Viramate author who gives more explanations (but nothing new from the screenshot I linked above).

**Comment by kusyarudaora123** [comment]
If the recovery method of the AP is as it is, it may eventually become a target of detection.

I would be grateful if you could recover from the item column.

**Comment by Masuzu** [comment]
Dear all,

You will find below some answers.

# The guarantees

Zooey Bot, Sarasa Bot and the Poker Bot do not modify the DOM of the game when they are used on their own, unless you use other extensions such as Viramate (depending on the settings) and Adblocker (some elements are blocked sometimes).
GBF Combat Sniffer does not modify either the DOM unless you set the parameter SynchronizeAbilityCooldowns to true.

Technically you can't know if a Chrome extension is installed or not if its ID is not fixed, which is the case for Zooey Bot, Sarasa Bot. If you want to do so, a brute force approach is required and would take thousands of years to complete on a supercomputer in this day and age.

# The future roadmap

1. Provide as soon as possible a different AP and BP refill system

2. Displeased about the current state and future direction the game is taking? Invested a lot of time and cash in an account which could be terminated on a whim or by mistake? Future projects still revolve around multimedia entertainment, are not completely unrelated to Granblue Fantasy, but will put a strong emphasis on providing a consumer-friendly gaming experience for every kind of player and offering experiences where ping advantage is not one of the main factors to "win".

Want to play using a console to be a DPS hero or have your own super macros to have an edge? Go ahead, laissez-faire, laissez-passer: everyone should enjoy the game in his or her own unique way, and special gaming modes to cater to these players should exist. Want a more chill gameplay as you mainly play on your phone or do not have the best ping? We should have other modes favoring this kind of gameplay and where everyone is on an equal footing. Want some new fancy things, ask, contribute and get them. Hopefully I (we?) can make good progress (together) on this front in the future days and weeks :)

# The speculation

I got some experimental test accounts banned for a week where the number of requests sent to the server during raids were abnormally high.
Other test accounts which were actively using Viramate or the GBF Combat Sniffer shortcuts and quick raid joining and botting and virtual machines located in some other country are fine.

Regarding the "warning message" carpet bombing, you get the idea from distribution lists work: you have a bunch of emails, and send to each of them mails by batch. That is my take on yesterday's and future in-game warning messages.

On a more personal note, I do feel like the community outrage is perfectly justified and ["fridge" ban](https://twitter.com/msh__121229/status/1117739778995134465) is hilariously sad.

Stay safe

**Comment by SunStrider8420** [comment]
Thanks for the update Masuzu, I look forward to the updates.

**Comment by GBFUSER1993** [comment]
Please made the refilling update ASAP, best regards!

**Comment by i-Grey** [comment]
Thank you for your info Masuzu!

**Comment by niayyy** [comment]
Got banned for 7 days.
Had been using zooey bot to farm the latest events for 22 boxes. Farming like 2 hrs and stopped for 1-2hrs repeatedly.
At the same time, yes, with viramate, adblock and sometimes google translate to translate the text of the story.
Anyone using zooey bot for the last event not got banned? really glad to hear that if so, this bot helps me a lot..

**Comment by GBFUSER1993** [comment]
@niayy yes, zooey bot alone without any other plugins for 48 box last event

**Comment by GBFUSER1993** [comment]
So the main problem is viramate I think

**Comment by niayyy** [comment]
> @niayy yes, zooey bot alone without any other plugins for 48 box last event

thx mate good to hear that

**Comment by UminekoScarlet** [comment]
Just to make sure I get your post correctly (as it directly involves the security of my account):
"I got some experimental test accounts banned for a week where the number of requests sent to the server during raids were abnormally high."
This is not the result of ZooeyBot, right? So basically nothing has changed for ZooeyBot users, aside from no longer being able to use Viramate features?

And to also answer niayyy: I 20 boxed last event (just like I botted all others before that) with ZooeyBot and I did not get banned.
I did not use Viramate while botting, but I always used it on my main machine I play GBF on.

**Comment by BB951** [comment]
@UminekoScarlet I believe there is still a slight risk using zooey, as the current ap refill system might be detectable. It might be safer to just run on 999ap and refill to be safe.

**Comment by Sundaerae** [comment]
@Masuzu
here's the patents on CyGames' anti-cheat. Don't really know how to read it since the wording is pretty weird since it was translated from Japanese. Hope it helps.
https://patents.google.com/patent/JP6473259B1/en?oq=6473259

**Comment by Masuzu** [comment]
@UminekoScarlet This was the result of tests unrelated to any of the bots.

@Sundaerae Thank you for the link, interesting indeed.

**Comment by jsun1234** [comment]
So are we free to go ahead and run it or should we wait until you investigate this new patent?

**Comment by Masuzu** [comment]
The patent just describes a method to replay recorded user events on virtual machines with the game also running, and to check that the server response obtained from those replayed actions match the recorded server logs.
It does not affect whatsoever the current bit which move the mouse and perform actual clicks. On a side note, I am sure everyone is very happy to see that their money goes into this instead of making the UI palatable.

**Comment by Masuzu** [comment]
Oh, and something which completely invalidate this method is browser zoom. There is no way to get reliably that information reliably, you may have something working one day and break the following day due to a browser update, or even changing your display resolution.

**Comment by zefri7** [comment]
should we be using the bot?

**Comment by Masuzu** [comment]
Business as usual, nothing changed with regards to Zooey bot and company.

**Comment by Chololoco13** [comment]
I was using the bot a few hours but I recharged the Ap manually (999 ap full) And host 30 raid.
I do not know if it's safe for the bot to reload quickly (automatically) but I do not want to risk it.
So far I have not had any problems.
I have the concern about how to recharge the ap
Google translator-

**Comment by Alissa1122** [comment]
Regarding to the refill concern, based on past experience, CY did record lots of logs and keep for a long time (said never delete), any evidence shows they already record such behavior? (send a special flag/package?)

based on viramate author’s comment, these detect changes should be able to be observed?

**Comment by Rkdheishins** [comment]
just for a little assurance ill be botting on my alt for the next 4 hours to hit capcha, as of right now its up and running with no warnings. will be back once capcha comes up

**Comment by Rkdheishins** [comment]
so capcha just kicked in right at 2 hours doing 1tk for gw, so good so far!

will be back if something comes up.
more then likely looks like will be fine with no adblocker/vira
thanks again masuzu for your hard long hours in making our gaming lives easier!

**Comment by dynameis** [comment]
> the game can now detect extensions, anything it doesn't recognize or is whitelisted = banned for 7 days
> 
> adblock seems safe, users only appear to be warned.

I'm pretty sure the game is not doing this,
I wrote a extension for my self and I'm sure It violate game rule
I use it without Warning messages  from 2018/12
plus I search the front end code , I can't find any extension detection code in it

**Comment by Poko302** [comment]
> I'm pretty sure the game is not doing this,
> I wrote a extension for my self and I'm sure It violate game rule
> I use it without Warning messages from 2018/12
> plus I search the front end code , I can't find any extension detection code in it
> 
> (I'm not zooey user , just passing by)

Just wondering, May I ask what did your extension do that u said it violate game rule?

**Comment by dynameis** [comment]
> Just wondering, May I ask what did your extension do that u said it violate game rule?

quest / event  botting

**Comment by justadeveloper565** [comment]
quest / event botting , but I only use it when I using my PC
if any validation (CAPTCHA, validate button) came out , extension only make beep sound ,
and I input CAPTCHA manually

**Comment by omieqqe** [comment]
Ok. After i take a risk and send email to Cy. Here is thier answer.
TLDR: They spotted i use modification UI (VM or Adblock)
=======
Here is the full mail. ( In Japanese)

グランブルーファンタジー運営事務局でございます。

 本メールにつきまして、複数回お問い合わせいただいていた場合、重ねてお伝えするようなご案内となる場合もございます。
 何とぞご了承いただきますようお願いいたします。

 お問い合わせいただきましたアカウントにつきまして、運営事務局にて調査をおこなったところ、不正ツールを利用してUIや一部動作の改変を行っている事を検知いたしました。

 グランブルーファンタジーTOPページにある「違反行為に対する措置について」に記載しておりますとおり、不正ツールを利用しているアカウントは制限措置の対象となります。
 そのため、該当のアカウントは「2019/04/22 19：00」までアカウント停止措置が行われております。

 なお、アカウント停止措置に関する調査内容の詳細は、個別では案内しておらず、質問も受け付けておりません。

 今後、本件に関するお問い合わせをいただきましてもご案内いたしかねます。
===============
Good luck ;) I take a vacation until 22.
@Masuzu

**Comment by ErogeSeiyuu** [comment]
"なお、アカウント停止措置に関する調査内容の詳細は、個別では案内しておらず、質問も受け付けておりません。

今後、本件に関するお問い合わせをいただきましてもご案内いたしかねます。"

it reads like they carpet-banned en-masse and will not lift ban for anything or anyone no matter what proofs people can provide.

**Comment by omieqqe** [comment]
> "なお、アカウント停止措置に関する調査内容の詳細は、個別では案内しておらず、質問も受け付けておりません。
> 
> 今後、本件に関するお問い合わせをいただきましてもご案内いたしかねます。"
> 
> it reads like they carpet-banned en-masse and will not lift ban for anything or anyone no matter what proofs people can provide.

Except you are famous or relate with Cygame *cough cough *

**Comment by Masuzu** [comment]
The Action Point refill was updated in https://gbtools.azurewebsites.net/ZooeyBot/en/Home/PatchNotes#3-7-0

**Comment by ErogeSeiyuu** [comment]
![unknown-17](https://user-images.githubusercontent.com/28994631/56317915-f1d0ed00-612b-11e9-8e6f-5f1e2671d963.png)
The acc posting this on Discord is a whale who's selling his whale acc for $5000 on acc trading discord.

**Comment by ErogeSeiyuu** [comment]
Time to bot 6 days a month and 3.5 hours per day.

**Comment by ghosimage1** [comment]
@ErogeSeiyuu ,
With the statement you provided we are safe to assume as long as we do not bot more than 6 days a month and 3.5 hours day, we are literally off their radar?
@Masuzu 
Thank you for all the effort in helping us! To make our life easier. Cheers!

**Comment by rockysilver** [comment]
The picture from Eroge is kind of blurry, so I can't read the whole thing clearly, but in Chinese that person is saying that "anyone played more than 4 hours a day for more than 7 days in a month will be subjected to the review".

Instead of just playing more than 4 hours a day or 7 days a month; because I would assume most active player will be playing more than 7 days a month.

**Comment by justadeveloper565** [comment]
source is here https://sora.komica.org/00/pixmicat.php?res=14437480
but I think that man is FAKE cygames employee
![image](https://user-images.githubusercontent.com/49720740/56342926-ecd37280-61eb-11e9-9a8b-4b2d9daa6d77.png)

**Comment by Masuzu** [comment]
It may very well be fake indeed.

**Comment by Masuzu** [comment]
New project, hoping to bring a MVP in the upcoming weeks :)
https://www.youtube.com/watch?v=kfJQEEJix50

**Comment by Sundaerae** [comment]
what's this for? testing grid?

**Comment by ErogeSeiyuu** [comment]
I bet 4 gislas that it's granblue first private server

**Comment by Rkdheishins** [comment]
wait,wait,wait this new project your working on, allows us to do exactly just that in game?

or was everything fake for some odd kinda testing purposes?

**Comment by YumeHaru** [comment]
> New project, hoping to bring a MVP in the upcoming weeks :)
> https://www.youtube.com/watch?v=kfJQEEJix50

I don't think toying with skins is a good idea especially when there will be idiots streaming that because it's cool yet it's what brought unwanted attention to Viramate (features that japanese don't have) but I do like the improved UI.
I would like a way to code more simply my bot skill routine like simple commands to drag and drop any special command like casting a skill on someone than the caster could be written by the user himself.

**Comment by Hildibrand** [comment]
> The picture from Eroge is kind of blurry, so I can't read the whole thing clearly, but in Chinese that person is saying that "anyone played more than 4 hours a day for more than 7 days in a month will be subjected to the review".
> 
> Instead of just playing more than 4 hours a day or 7 days a month; because I would assume most active player will be playing more than 7 days a month.

What does this mean, though? They will manually review any player logs who played more than 4 hours daily for more than 7 days in the same month? Because I figure that would be a lot of work to do and don't think they manage to do that.
The "random banwaves" to anyone who has been banned before also seems like a dumb strategy. Tbh a random post on 4chan claiming to be a Cygames member it's not really to be trusted.

**Comment by ErogeSeiyuu** [comment]
> They will manually review any player logs who played more than 4 hours daily for more than 7 days in the same month? Because I figure that would be a lot of work to do and don't think they manage to do that.

Random screening. Not all of them. Because there's tons of people who use both viramate and bot but not get banned, and there's people who did. Same with random banwaves. It's much easier to do random bans of people who has previous ban history. 

It can be (and probably is) fake, but nobody really knows when the ban will hit. I've never been hit with a ban despite using both for nearly 2 years. However I was previously banned for casinobot twice. So my confirmation bias feels that it applies to me to some extent.

Yall are free to believe w/e you wish. I will still bot on as usual anyway. If I get temp banned again next GW I'll just sell the acc off, and if I get permanently banned, then good riddance and I can move forward to play other things, and clean up my desk for once since I won't need to keep my old laptop next to my main rig to keep track on zoibot anymore.

**Comment by rockysilver** [comment]
> > The picture from Eroge is kind of blurry, so I can't read the whole thing clearly, but in Chinese that person is saying that "anyone played more than 4 hours a day for more than 7 days in a month will be subjected to the review".
> > Instead of just playing more than 4 hours a day or 7 days a month; because I would assume most active player will be playing more than 7 days a month.
> 
> What does this mean, though? They will manually review any player logs who played more than 4 hours daily for more than 7 days in the same month? Because I figure that would be a lot of work to do and don't think they manage to do that.
> The "random banwaves" to anyone who has been banned before also seems like a dumb strategy. Tbh a random post on 4chan claiming to be a Cygames member it's not really to be trusted.

The chance is this is nothing, it is very likely that the post is fake. If they have a reliable way to catch bot, we would all be perma banned right now instead of a warning.

**Comment by Masuzu** [comment]
Small updates, the tedious part is now nailed down. Time for the big guns.
https://www.youtube.com/watch?v=_d1M5Q5eouA

**Comment by devvah** [comment]
Ehh you know these stories about modders spending x years to make fan game or something and then getting DMCA'd near release? Be smarter and make what LoL is to DotA rather than being at mercy of hosts respecting claims as they definitely will.

**Comment by ErogeSeiyuu** [comment]
It's gonna be even worse in gbf case since it is a live service and a big, profitable one at that.

Cygames are going to crack down on this so hard.

**Comment by i-Grey** [comment]
I don't know what did you think about but I don't interest at all. 
What would be fun by playing on the standalone server. If you insist on doing to you would got crack down from Cygames for sure.

For me myself, and I'm also confident that almost people in this forum would have the same opinion.
We would really appreciate if you keep ZooeyBot up to date. That's enough.
